src/EnergyPlus/ScheduleManager.cc

```diff
@@ -216,17 +216,22 @@ namespace Sched {
         }
     } // ScheduleDay::populateFromHrMinVals()
 
-    ScheduleConstant *AddScheduleConstant(EnergyPlusData &state, std::string const &name)
+    ScheduleConstant *AddScheduleConstant(EnergyPlusData &state, std::string const &name, Real64 value)
     {
         auto const &s_sched = state.dataSched;
+        auto const &s_glob = state.dataGlobal;
 
         auto *sched = new ScheduleConstant;
         sched->Name = name;
+        sched->type = SchedType::Constant;
         sched->Num = (int)s_sched->schedules.size();
+        sched->currentVal = value;
+        // When InitConstantScheduleData is called, TimeStepsInHour is 0, so we ensure 24
+        sched->tsVals.assign(Constant::iHoursInDay * max(1, s_glob->TimeStepsInHour), value);
```

Comment:
Seems like a reasonable improvement to protect the setup of this stuff in unit testing.

Reply:
I don’t think this is robust. The IDF file will probably set the TimeStep to something greater than 1, at which point the simulation may want to grab values at indices that don’t exist in this array. This array should be resized after the actual time step of the simulation is known.

Reply:
1) In the GetInput routine, it will be resized.
2) Unit tests that would have this issue would be a) no GetInput called and b) A call to getDayVals is happening